examples/conversation_with_RAG_agents/rag_example.py

```diff
@@ -1,63 +1,146 @@
-                "config_name": "qwen_emb_config",
-                "model_name": "text-embedding-v2",
-                "api_key": f"{os.environ.get('DASHSCOPE_API_KEY')}",
-            },
-        ],
-    )
+    # prepare html for api agent
+    prepare_docstring_html()
+
+    # prepare models
+    with open("configs/model_config.json", "r", encoding="utf-8") as f:
+        model_configs = json.load(f)
 
-    with open("./agent_config.json", "r", encoding="utf-8") as f:
+    # load config of the agents
+    with open("configs/agent_config.json", "r", encoding="utf-8") as f:
         agent_configs = json.load(f)
-    tutorial_agent = LlamaIndexAgent(**agent_configs[0]["args"])
-    code_explain_agent = LlamaIndexAgent(**agent_configs[1]["args"])
-    summarize_agent = DialogAgent(**agent_configs[2]["args"])
+
+    agent_list = agentscope.init(
+        model_configs=model_configs,
+        agent_configs=agent_configs,
+    )
+    rag_agent_list = agent_list[:4]
+    guide_agent = agent_list[4]
+
+    # the knowledge bank can be configured by loading config file
+    knowledge_bank = KnowledgeBank(configs="configs/knowledge_config.json")
+
+    # alternatively, we can easily input the configs to add data to RAG
+    knowledge_bank.add_data_as_knowledge(
+        knowledge_id="agentscope_tutorial_rag",
+        emb_model_name="qwen_emb_config",
+        data_dirs_and_types={
+            "../../docs/sphinx_doc/en/source/tutorial": [".md"],
+        },
+    )
+
+    # let knowledgebank to equip rag agent with a (set of) knowledge
+    # corresponding to its knowledge_id_list
+    for agent in rag_agent_list:
+        knowledge_bank.equip(agent, agent.knowledge_id_list)
+
+    # an alternative way is to provide knowledge list to agents
+    # when initializing them one by one, e.g.
+    #
+    # ```
+    # knowledge = knowledge_bank.get_knowledge(knowledge_id)
+    # agent = LlamaIndexAgent(
+    #   name="rag_worker",
+    #   sys_prompt="{your_prompt}",
+    #   model_config_name="{your_model}",
+    #   knowledge_list=[knowledge], # provide knowledge object directly
+    #   similarity_top_k=3,
+    #   log_retrieval=False,
+    #   recent_n_mem_for_retrieve=1,
+    # )
+    # ```
+
+    rag_agent_names = [agent.name for agent in rag_agent_list]
+    # update guide agent system prompt with the descriptions of rag agents
+    rag_agent_descriptions = [
+        "agent name: "
+        + agent.name
+        + "\n agent description："
+        + agent.description
+        + "\n"
+        for agent in rag_agent_list
+    ]
+
+    guide_agent.sys_prompt = (
+        guide_agent.sys_prompt
+        + AGENT_CHOICE_PROMPT.format(
+            "".join(rag_agent_descriptions),
+        )
+    )
 
     user_agent = UserAgent()
-    # start the conversation between user and assistant
     while True:
+        # The workflow is the following:
+        # 1. user input a message,
+        # 2. if it mentions one of the agents, then the agent will be called
```

Comment:
Should we tell the user that the word `mention` referred to the `@` operation here?